internal/repository/sync_test.go

```diff
@@ -0,0 +1,493 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+	remoteRef, err = repoRemote.Reference(refNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, localRef.Hash(), remoteRef.Hash())
+
+	localAnotherRef, err := repoLocal.r.Reference(anotherRefNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	remoteAnotherRef, err := repoRemote.Reference(anotherRefNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, localAnotherRef.Hash(), remoteAnotherRef.Hash())
+
+	localRSLRef, err = repoLocal.r.Reference(rslRefNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	remoteRSLRef, err = repoRemote.Reference(rslRefNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, localRSLRef.Hash(), remoteRSLRef.Hash())
+
+	localPolicyRef, err = repoLocal.r.Reference(policyRefNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	remotePolicyRef, err = repoRemote.Reference(policyRefNameTyped, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, localPolicyRef.Hash(), remotePolicyRef.Hash())
+}
+
+func TestPull(t *testing.T) {
+	remoteTmpDir, err := os.MkdirTemp("", "gittuf")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(remoteTmpDir) //nolint:errcheck
+
+	remoteR, err := git.PlainInit(remoteTmpDir, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	remoteRepo := &Repository{r: remoteR}
+	rootKeyBytes, err := os.ReadFile(filepath.Join("test-data", "root"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	targetsPubKeyBytes, err := os.ReadFile(filepath.Join("test-data", "targets.pub"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	targetsKeyBytes, err := os.ReadFile(filepath.Join("test-data", "targets"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.InitializeRoot(context.Background(), rootKeyBytes, false); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.AddTopLevelTargetsKey(context.Background(), rootKeyBytes, targetsPubKeyBytes, false); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.InitializeTargets(context.Background(), targetsKeyBytes, policy.TargetsRoleName, false); err != nil {
+		t.Fatal(err)
+	}
+	emptyTreeHash, err := gitinterface.WriteTree(remoteRepo.r, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	refName := "refs/heads/main"
+	anotherRefName := "refs/heads/feature"
+	remoteName := "origin"
+	commitID, err := gitinterface.Commit(remoteRepo.r, emptyTreeHash, refName, "Initial commit", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.RecordRSLEntryForReference(refName, false); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.r.Storer.SetReference(plumbing.NewSymbolicReference(plumbing.HEAD, plumbing.ReferenceName(refName))); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.r.Storer.SetReference(plumbing.NewHashReference(plumbing.ReferenceName(anotherRefName), commitID)); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.RecordRSLEntryForReference(anotherRefName, false); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
This looks like this is needed for some initial repo setup? Can this also be pulled into a common func?